public/js/companies/edit.js

```diff
@@ -0,0 +1,83 @@
+window.onload = (event) => {
+  const form = document.querySelector('.all-content-create-new-content-item-wrapper');
+  const badRequestError = document.getElementById('bad-request-error');
+  const unknownError = document.getElementById('unknown-error');
+
+  form.onsubmit = event => {
+    event.preventDefault();
+    badRequestError.style.display = unknownError.style.display = 'none';
+
+    const password = document.getElementById("new-password").value;
+    const credit_amount = parseInt(document.getElementById("credit-amount").value);
+    const company_id = document.getElementById("company_id").value
+
+    const data = {
+      company_id: company_id,
+      password: password,
+      credit_amount: credit_amount
+    }
+
+    let status; // 0 not decided -> do nothing, 1 password changed -> send new password, 2 password not changed -> do nothing
```

Comment:
Status olayından kurtul. Error kontrolünü direk return ile if lerin değerini değiştir. Zaten yorumlar ile de uyuşmuyor bazı yerler tuhaf olmuş. Ayrıca kodun bazı yerlerde doğru çalışmaz. Callback mantığını kullanmamamın bir sebebi var createConfirm'de. Sync bir dil olduğu için js callback leri kullanman gerektiği gibi kullanmazsan bir garantisi yok o kodun lineer okunacağının. O yüzden confirm yaratırsan içindeki callback de devam et işlemi yapmaya. ÇOK ÇOK ÇOK önemli

Reply:
Burada zaten parallel olmasının gücünü kullandım, çünkü confirmleri durumlara spesifik olarak çkarmak istiyorum. Bunu daha düzgün nasıl yapabilirim noktasında ışık tutabilirsen sevinirim. Atıyorum, eğer şifreye adam bir şey yazmadıysa, şifrenin checkini atmasın ona göre de confirm field'ı çıkarmasın. O fieldları opsiyonel yapmaya çalışmaktı amacım.

Reply:
Tekrar bakınca buraya çok daha kolay bir şey yazdım şimdi